### ISSUE TITLE
Display most common trips

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-server#88]
@Mogeng wrote a script that does trajectory matching to find the top 10 "hot" routes for a particular user. It would be great to make this available as part of the user results.

The steps to do so would involve:
- Writing automated tests for the existing trajectory matching to ensure that it works properly
- Adding an API endpoint that will return the routes
- Adding an map interface to display the routes as part of the user results. This can be done using the google javascript APIs, or leaflet.js. Note that I haven't used leaflet.js before, but they claim that it works well on all mobile browsers.

**Comment by jimmydief** [comment]
@zacatac, @lechiang and I would like to claim this.

**Comment by shankari** [comment]
So the code to return the common routes for a user is here:
https://github.com/e-mission/e-mission-server/blob/master/CFC_WebApp/main/route_matching.py#L455

You can verify that they are calculated correctly for your data by running the corresponding plot method from a python shell (ipython/ipython notebook).
https://github.com/e-mission/e-mission-server/blob/master/CFC_WebApp/main/route_matching.py#L468

This seems to assume that the clusters have been updated correctly.
Let me look to see how to do that...

**Comment by shankari** [comment]
@Mogeng, can you clarify where `update_user_routeClusters` is invoked? In particular, what are the clusters that are passed in to it? Is there a script that can be run to update them?

**Comment by Mogeng** [comment]
Let me check it~

**Comment by Mogeng** [comment]
These are the function in route_matching, can be invoked in the profile.py or anywhere that runs regularly. I tested these function in my test_simple.py, which was discarded from github because of the privacy issues. I still have this test_simple.py. Should I pass it to you?

Thanks,
Mogeng

**Comment by shankari** [comment]
Yeah I saw that they are in `route_matching.py`, but `update_user_clusters` needs to be invoked periodically in the background, right? Where do you get the `clusters` parameter that you pass in to it?
Can you invoke it from a script that runs periodically?
`profile.py` is fine.

**Comment by Mogeng** [comment]
Yes, that is no problem, so you want me to push the changes in profile.py? Or I just sent updated profile.py to you?

Just one more thing, we need to make sure there is a current_db.Stage_routeDistanceMatrix and current_db.Stage_routeCluster.


Mogeng

**Comment by shankari** [comment]
Push the changes! You need to boost your github commit score :)

**Comment by Mogeng** [comment]
Not sure why but when I push, Permission to e-mission/e-mission-server.git denied to Mogeng.

Thanks
Mogeng

**Comment by shankari** [comment]
That's because it is not a shared repo. You need to submit a pull request.

Do you remember the flow for the pull request that we saw last time (fork
your own copy, clone, make changes, push to your fork, submit pull request
to main...)?

If not, just email me the Profile file and I'll make sure that it gets in...

Shankari

**Comment by Mogeng** [comment]
I didn’t make it…So I attached the Profile.py here.

Mogeng
